packages/jest-haste-map/src/index.js

```diff
@@ -291,7 +290,21 @@ class HasteMap extends EventEmitter {
    * 1. read data from the cache or create an empty structure.
    */
   read(): InternalHasteMap {
-    return this._parse(fs.readFileSync(this._cachePath, 'utf8'));
+    if (v8.deserialize) {
+      // This may throw. `_buildFileMap` will catch it and create a new map.
+      const {version, hasteMap} = v8.deserialize(
+        fs.readFileSync(this._cachePath),
+      );
+      if (version !== process.versions.v8) {
+        throw new Error('jest-haste-map: v8 versions do not match.');
```

Comment:
We should be smarter here, like purging the cache and keep it going, but for the moment it's OK.

Reply:
It does already do that. Exceptions from this function are caught in the parent and a new map is created which will overwrite this one. For all intents and purposes, this error message is not user visible at this time.